### ISSUE TITLE
PKGEXT environment variable is ignored

### ISSUE THREAD

**Issue by gagarski** [rmarquis/pacaur#667]
###### Version

```
pacaur 4.7.5
```

###### Description

The documentation says 

> In addition, makepkg environment variables are fully honored. See makepkg(8).

I am trying to avoid compressing and decompressing packages while updating my system by setting PKGEXT to ".tar".

However, it seems like PKGEXT variable is ignored by pacaur .
That's how i am trying to do it:
```
$ which pacaur
pacaur: aliased to PKGEXT=.tar BUILDDIR="/tmp/pacaur-tmp-${USER}" AURDEST="/tmp/pacaur-tmp-${USER}" pacaur

$ pacaur -S intellij-idea-ultimate-edition
```

PKGEXT and AURDEST variables seem to work fine, but pacaur still compresses the package during build. I can see continuous "Compressing package" stage during build and "*.tar.xz" files appearing in build directory.
```
$ ls -la /tmp/pacaur-tmp-gagarski/intellij-idea-ultimate-edition/
total 641480
drwxr-xr-x 5 gagarski users       220 Mar 13 13:29 .
drwx------ 3 gagarski users        60 Mar 13 13:27 ..
drwxr-xr-x 8 gagarski users       280 Mar 13 13:30 .git
-rw-r--r-- 1 gagarski users 527841445 Mar 13 13:28 ideaIU-2016.3.5-no-jdk.tar.gz
-rw-r--r-- 1 gagarski users       137 Mar 13 13:27 intellijidea.sh
-rw-r--r-- 1 gagarski users 129015808 Mar 13 13:30 intellij-idea-ultimate-edition-2016.3.5-1-any.pkg.tar.xz
-rw-r--r-- 1 gagarski users       246 Mar 13 13:27 jetbrains-idea.desktop
drwxr-xr-x 3 gagarski users        60 Mar 13 13:28 pkg
-rw-r--r-- 1 gagarski users      1881 Mar 13 13:27 PKGBUILD
drwxr-xr-x 3 gagarski users       120 Mar 13 13:28 src
-rw-r--r-- 1 gagarski users      1074 Mar 13 13:27 .SRCINFO
```

###### Output

`pacaur -S intellij-idea-ultimate-edition`

```
$ pacaur -S intellij-idea-ultimate-edition
:: Package intellij-idea-ultimate-edition not found in repositories, trying AUR...
:: resolving dependencies...
:: looking for inter-conflicts...

AUR Packages  (1) intellij-idea-ultimate-edition-2016.3.5-1  

:: Proceed with installation? [Y/n] 

:: Retrieving package(s)...
Cloning into 'intellij-idea-ultimate-edition'...
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 6 (delta 0), reused 2 (delta 0)
Unpacking objects: 100% (6/6), done.
:: View intellij-idea-ultimate-edition PKGBUILD? [Y/n] 

:: intellij-idea-ultimate-edition PKGBUILD viewed
:: Checking intellij-idea-ultimate-edition integrity...
==> Making package: intellij-idea-ultimate-edition 2016.3.5-1 (Mon Mar 13 13:27:14 MSK 2017)
==> Retrieving sources...
  -> Downloading ideaIU-2016.3.5-no-jdk.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   154  100   154    0     0    655      0 --:--:-- --:--:-- --:--:--   655
100  503M  100  503M    0     0  7171k      0  0:01:11  0:01:11 --:--:-- 8591k
  -> Found intellijidea.sh
  -> Found jetbrains-idea.desktop
==> Validating source files with sha256sums...
    ideaIU-2016.3.5-no-jdk.tar.gz ... Passed
    intellijidea.sh ... Passed
    jetbrains-idea.desktop ... Passed
:: Building intellij-idea-ultimate-edition package(s)...
==> Making package: intellij-idea-ultimate-edition 2016.3.5-1 (Mon Mar 13 13:28:31 MSK 2017)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Entering fakeroot environment...
==> Starting package()...
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Compressing man and info pages...
==> Checking for packaging issue...
==> Creating package "intellij-idea-ultimate-edition"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: intellij-idea-ultimate-edition 2016.3.5-1 (Mon Mar 13 13:33:46 MSK 2017)
==> Cleaning up...
:: Installing intellij-idea-ultimate-edition package(s)...
loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) intellij-idea-ultimate-edition-2016.3.5-1

Total Installed Size:  960.57 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                                                                                                                     [#####################################################################################################] 100%
(1/1) checking package integrity                                                                                                                                   [#####################################################################################################] 100%
(1/1) loading package files                                                                                                                                        [#####################################################################################################] 100%
(1/1) checking for file conflicts                                                                                                                                  [#####################################################################################################] 100%
(1/1) checking available disk space                                                                                                                                [#####################################################################################################] 100%
:: Processing package changes...
(1/1) installing intellij-idea-ultimate-edition                                                                                                                    [#####################################################################################################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Updating the desktop file MIME type cache...
```

###### Debug output

```
rm -rf /tmp/pacaur-tmp-gagarski  # Cleaning cache to start everything over
PKGEXT=.tar BUILDDIR="/tmp/pacaur-tmp-${USER}" AURDEST="/tmp/pacaur-tmp-${USER}" bash -x pacaur -S intellij-idea-ultimate-edition # Since I use ZSH, bash does not respect my aliases, so here I am passing my variables to bash manually 
```

[Gist with debug output](https://gist.github.com/gagarski/e473ebc71e5296bb07c4e45764aaf483)

###### Playing with makepkg

As your "Bug reporting" page suggests, I tried to do the same with `makepkg`.

This command creates compressed package:
```
PKGEXT=.tar makepkg -o && makepkg -es 
```

However, this command creates uncompressed tar package:
```
PKGEXT=.tar makepkg -o && PKGEXT=.tar makepkg -es
```

**Comment by rmarquis** [comment]
Thx for the report. Indeed, if the var isn't exported it isn't passed to makepkg and is overwritten when makepkg.conf is sourced.

Fix seems easy to implement, I'll double check that all env variables work before pushing tonight.

**Comment by rmarquis** [comment]
Fixed in 6803b7b92fd682b2d06830f118d66407806b6410.
